internal/pkg/gobgp/interface.go

```diff
@@ -0,0 +1,198 @@
+					lsNode.SrgbEnd = srCapabilities[0].GetEnd()
+				}
+			}
+			tedElems = append(tedElems, lsNode)
+
+		case *api.LsLinkNLRI:
+			// Get information from MP-REACH-NLRI Attr
+			localNodeId := typedLinkStateNlri.GetLocalNode().GetIgpRouterId()
+			localNodeAsn := typedLinkStateNlri.GetLocalNode().GetAsn()
+			remoteNodeId := typedLinkStateNlri.GetRemoteNode().GetIgpRouterId()
+			remoteNodeAsn := typedLinkStateNlri.GetRemoteNode().GetAsn()
+			localIP := net.ParseIP(typedLinkStateNlri.GetLinkDescriptor().GetInterfaceAddrIpv4())
+			remoteIP := net.ParseIP(typedLinkStateNlri.GetLinkDescriptor().GetNeighborAddrIpv4())
+
+			localNode := table.NewLsNode(localNodeAsn, localNodeId)
+			remoteNode := table.NewLsNode(remoteNodeAsn, remoteNodeId)
+			lsLink := table.NewLsLink(localNode, remoteNode)
+			lsLink.LocalIP = localIP
+			lsLink.RemoteIP = remoteIP
+
+			// Get information from BGP-LS Attr
+			for _, pathAttr := range pathAttrs {
+				typedPathAttr, err := pathAttr.UnmarshalNew()
+				if err != nil {
+					return nil, err
+				}
+				if bgplsAttr, ok := typedPathAttr.(*api.LsAttribute); ok {
+					igpMetric := table.NewMetric(table.MetricType(table.IGP_METRIC), bgplsAttr.GetLink().GetIgpMetric())
+					lsLink.Metrics = append(lsLink.Metrics, igpMetric)
+
+					lsLink.AdjSid = bgplsAttr.GetLink().GetSrAdjacencySid()
+				}
+			}
+			tedElems = append(tedElems, lsLink)
+		case *api.LsPrefixV4NLRI:
+			// Get information from MP-REACH-NLRI Attr
+			var sidIndex uint32
+			for _, pathAttr := range pathAttrs {
+				typedPathAttr, err := pathAttr.UnmarshalNew()
+				if err != nil {
+					return nil, err
+				}
+				// Get information from BGP-LS Attr
+				if bgplsAttr, ok := typedPathAttr.(*api.LsAttribute); ok {
+					// if sidIndex != 0 then Loopback interface Address Prefix
+					sidIndex = bgplsAttr.GetPrefix().GetSrPrefixSid()
+				}
+				if mpReachNlriAttr, ok := typedPathAttr.(*api.MpReachNLRIAttribute); ok {
+					for _, nlri := range mpReachNlriAttr.GetNlris() {
+						typedNlri, err := nlri.UnmarshalNew()
+						if err != nil {
+							return nil, err
+						}
+						if lsNlri, ok := typedNlri.(*api.LsAddrPrefix); ok {
+
+							prefNlri, err := lsNlri.GetNlri().UnmarshalNew()
+							if err != nil {
+								return nil, err
+							}
+							prefv4Nlri := prefNlri.(*api.LsPrefixV4NLRI)
+							localNodeId := prefv4Nlri.GetLocalNode().GetIgpRouterId()
+							localNodeAsn := prefv4Nlri.GetLocalNode().GetAsn()
+							prefixV4 := prefv4Nlri.GetPrefixDescriptor().GetIpReachability()
+
+							localNode := table.NewLsNode(localNodeAsn, localNodeId)
+							lsPrefixV4 := table.NewLsPrefixV4(localNode)
+							lsPrefixV4.SidIndex = sidIndex
+							if len(prefixV4) != 1 {
+								return nil, errors.New("invalid prefix length")
+							}
+							_, lsPrefixV4.Prefix, _ = net.ParseCIDR(prefixV4[0])
+							tedElems = append(tedElems, lsPrefixV4)
+						}
+					}
+				}
+			}
+		}
+	default:
+		fmt.Printf("invalid Nlri Type: %T\n", typedNlri)
```

Comment:
This line can be deleted because output to stdout should be done only in the logger. In that case, include typedNlri in errors.New() on the next line.